server/app/boards.go

```diff
@@ -210,6 +210,14 @@ func (a *App) DuplicateBoard(boardID, userID, toTeam string, asTemplate bool) (*
 				},
 			})
 		}
+		if attachmentID, ok := block.Fields["attachmentId"]; ok {
+			blockIDs = append(blockIDs, block.ID)
+			blockPatches = append(blockPatches, model.BlockPatch{
+				UpdatedFields: map[string]interface{}{
+					"attachmentId": attachmentID,
+				},
+			})
+		}
```

Comment:
Rather than duplicate the code. I would set the FieldID based on block type.
```
		fieldName := ""
		if block.Type == model.TypeImage {
			fieldName = "fileId"
		} else if block.Type == model.TypeAttachment {
			fieldName = "attachmentID"
		}
		if fieldID, ok := block.Fields[fieldName]; ok {
		...
```

Reply:
Okay, will change it